app/Models/Game.php

```diff
@@ -160,6 +160,28 @@ public function shouldBeSearchable(): bool
         return false;
     }
 
+    // == temp subset detection
+
+    public function getParentGameId(): ?int
+    {
+        // Use regular expression to check if the title includes a subset pattern and extract the base title.
+        if (preg_match('/(.+)\[Subset - .+\]/', $this->Title, $matches)) {
+            // Trim to ensure no leading/trailing spaces.
+            $baseSetTitle = trim($matches[1]);
+
+            // Attempt to find a game with the base title and the same console ID.
+            $game = Game::where('Title', $baseSetTitle)
+                ->where('ConsoleID', $this->ConsoleID)
+                ->first();
+
+            // If a matching game is found, return its ID.
+            return $game?->id ?? null;
```

Comment:
This should probably return the whole game model, as both existing calls to `getParentGameIdFromGameTitle` proceed to fetch additional information from the related record.

Reply:
Changing this to return a collection rather than an int will require refactoring all the areas that take this as input since they are only expecting a game id not a whole collection. Would that be better as a seperate PR?

Reply:
I'm seeing things like this:
```
    $parentGameId = getParentGameIdFromGameTitle($game->Title, $game->system_id);
    if ($parentGameId) {
        $parentGame = Game::firstWhere('ID', $parentGameId);
```

Which could just be:
```
    $parentGame = getParentGameIdFromGameTitle($game->Title, $game->system_id);
    if ($parentGame) {
```

And:
```
    $parentGameId = getParentGameIdFromGameTitle($gameDataOut['Title'], $gameDataOut['ConsoleID']);

    $query = "SELECT players_total AS NumDistinctPlayers FROM GameData WHERE ID=" . ($parentGameId ?? $gameID);
    $gameMetrics = legacyDbFetch($query);
    $gameDataOut['ParentGameID'] = $parentGameId;
    $gameDataOut['NumDistinctPlayers'] = $gameMetrics['NumDistinctPlayers'] ?? 0;
```

Could be:
```
    $parentGame = getParentGameIdFromGameTitle($gameDataOut['Title'], $gameDataOut['ConsoleID']);
    $gameDataOut['ParentGameId'] = $parentGame?->id;
    $gameDataOut['NumDistinctPlayers'] = $parentGame?->players_total ?? 0;
```
 
In the other three cases where only the ID is necessary, just pull it from the model: `$parentGame?->id`.

I don't think these changes are complex enough to require a separate PR, but if you don't feel comfortable making the changes, someone else could do them later.

Reply:
These suggestions have been applied to latest.